payments/providers/talpa_ecom.py

```diff
@@ -0,0 +1,445 @@
+        self.user_id = payment_dict.get("userId", "")
+        self.status = payment_dict.get("status", "")
+        self.payment_method = payment_dict.get("paymentMethod", "")
+        self.payment_type = payment_dict.get("paymentType", "")
+        self.total_excl_tax = int(payment_dict.get("totalExclTax", 0))
+        self.total = int(payment_dict.get("total", 0))
+        self.tax_amount = int(payment_dict.get("taxAmount", 0))
+        self.additional_info = payment_dict.get("additionalInfo", "")
+        self.description = payment_dict.get("description", None)
+        self.token = payment_dict.get("status", "")
+        self.timestamp = payment_dict.get("status", "")
+        self.payment_method_label = payment_dict.get("paymentMethodLabel", "")
+
+
+class TalpaEComProvider(PaymentProvider):
+    """Talpa eCommerce Platform specific integration utilities and configuration."""
+
+    ui_return_url_param_name = "VENE_UI_RETURN_URL"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.url_order_experience_api = self.config.get(
+            VENE_PAYMENTS_TALPA_ECOM_ORDER_API_URL
+        )
+        self.url_payment_experience_api = self.config.get(
+            VENE_PAYMENTS_TALPA_ECOM_PAYMENT_API_URL
+        )
+
+        checkout_url = self.config.get(VENE_PAYMENTS_TALPA_ECOM_CHECKOUT_URL, "")
+        if not checkout_url.endswith("/"):
+            checkout_url = f"{checkout_url}/"
+
+        self.url_checkout = f"{checkout_url}{{talpa_ecom_id}}?user={{user_hash}}"
+
+    @staticmethod
+    def get_config_template() -> dict:
+        """Keys and value types what Bambora requires from environment"""
+        return {
+            VENE_PAYMENTS_TALPA_ECOM_PAYMENT_API_URL: (
+                str,
+                "https://checkout.api.hel.fi/v1/payment",
+            ),
+            VENE_PAYMENTS_TALPA_ECOM_ORDER_API_URL: (
+                str,
+                "https://checkout.api.hel.fi/v1/order",
+            ),
+            VENE_PAYMENTS_TALPA_ECOM_CHECKOUT_URL: (str, "https://checkout.hel.fi"),
+            VENE_PAYMENTS_TALPA_ECOM_API_NAMESPACE: (str, "venepaikat"),
+        }
+
+    def initiate_payment(self, order: Order) -> str:
+        """Initiate payment by creating an Order on the Talpa eCom service"""
+        language = (
+            order.lease.application.language
+            if hasattr(order, "lease") and order.lease.application
+            else settings.LANGUAGE_CODE
+        )
+        user_hash = get_customer_hash(order.customer)
+
+        headers = {"user": user_hash}
+
+        payload = {
+            "namespace": self.config.get(VENE_PAYMENTS_TALPA_ECOM_API_NAMESPACE),
+            "user": user_hash,
+        }
+
+        self.payload_add_customer(payload, order)
+        self.payload_add_products(payload, order, language)
+
+        try:
+            r = requests.post(
+                self.url_order_experience_api, json=payload, headers=headers, timeout=60
```

Comment:
Does our nginx frontend still have a 60 second timeout waiting for response from backend? The timeout in this request could be a few seconds shorter, so that the more detailed error reaches the end user?